### ISSUE TITLE
Data set processing

### ISSUE THREAD

**Issue by ly1035327995** [quantumxiaol/yolov8-small-target-detection#1]
Hello, there is no label information in the negative image set of NWPU data set. May I ask what to do when it is converted into yolo format data set?

**Comment by quantumxiaol** [comment]
For YOLO, if an image has no objects, you should create an empty label file for it. This way, YOLO will understand that there are no objects in the image.

In the NWPU dataset, the negative image set contains 150 images without any objects. You can name them as 651.jpg and create an empty label file 651.txt. When YOLO receives an empty label file, it will know that there are no objects inside.

**Comment by quantumxiaol** [comment]
ALL images are used during training. If no labels are found then the image simply has no labels.

'missing' labels mean that no labels.txt file is found, which is fine, the image simply has no objects. 'empty' means an empty labels.txt file was found. They both have the same result.

https://github.com/ultralytics/yolov5/discussions/7148

**Comment by ly1035327995** [comment]
What that means is that images that don't have labels create an empty.txt for them

**Comment by ly1035327995** [comment]
If I want to run nwpu on yolo, untagged pictures don't participate in training, right?

**Comment by quantumxiaol** [comment]
Yes, if you want to run the NWPU dataset on YOLO and ensure that untagged images (images without any objects) do not participate in training, you need to follow a specific process. Here’s how you can handle it:

No Label File: If an image does not contain any objects, you can omit the corresponding label file. YOLO will skip these images during the training process. This means the image will not be used in training if there is no .txt file associated with it.

Empty Label File: Alternatively, you can create an empty label file for each image without objects. YOLO will recognize the empty file and understand that there are no objects in that image. However, this will still include the image in the training dataset but with no targets to learn from.

**Comment by kkkkkkkomo** [comment]
你好，请问你提到修改了dota数据集标签计算方法具体是指什么呢，是dota-obb.py里的方法吗

**Comment by quantumxiaol** [comment]
最开始我使用dota_obb.py中的方法将旋转框转化为矩形框，然而新的标签框与原来的相比更小，丢了很多信息，后来使用dota.py中的方法来处理，相比原来的更大，你可以理解为一个是内接矩形一个是外接矩形

**Comment by quantumxiaol** [comment]
左边是dota_obb.py中的转换，有些问题，右边是dota.py的转换，我觉得本来目标就小，用个外接矩形更好


         原始邮件
         
       
发件人：kkkkkkkomo ***@***.***&gt;
发件时间：2025年3月17日 15:14
收件人：quantumxiaol/yolov8-small-target-detection ***@***.***&gt;
抄送：APLiu ***@***.***&gt;, Comment ***@***.***&gt;
主题：Re: [quantumxiaol/yolov8-small-target-detection] Data set processing(Issue #1)



kkkkkkkomo left a comment (quantumxiaol/yolov8-small-target-detection#1)

你好，请问你提到修改了dota数据集标签计算方法具体是指什么呢，是dota-obb.py里的方法吗

**Comment by kkkkkkkomo** [comment]
好的，感谢！
